### PR TITLE
Implement `in` for `list_type`

### PR DESCRIPTION
**Pull request by raxyte**
Implements support for finding a value inside a list with the following behavior: `x in y`
- `y == null` => `null` (Unimplemented operator)
- type mismatch => `null` + warning

Closes https://github.com/tenzir/issues/issues/2179

### CODE REVIEW

File: libtenzir/src/tql2/eval_binary.cpp [hunk 1]
```diff
@@ -434,6 +434,41 @@ struct EvalBinOp<ast::binary_op::in, string_type, string_type> {
   }
 };
 
+// TODO: Compare numbers across types
+// TODO: {} in [{}, ...]
+template <basic_type L>
+struct EvalBinOp<ast::binary_op::in, L, list_type> {
```
Comment (dominiklohmann) [inline libtenzir/src/tql2/eval_binary.cpp:440]: Did you consider implementing this in terms of `EvalBinOp<ast::binary_op::equal, …>`?

I think you can achieve this by slicing the left-hand side to single-element arrays, and getting the corresponding value-array from  the right-hand side for each entry (assuming it is not null or empty).

Comment (raxyte) [reply libtenzir/src/tql2/eval_binary.cpp:440]: You probably could do that, but then you need a build a long if-chain to dispatch the correct list type. Secondly, you'd need to copy (allocate) the output arrays into a single one.

File: libtenzir/src/tql2/eval_binary.cpp
Comment (jachris) [inline libtenzir/src/tql2/eval_binary.cpp:1]: Can you write some integration tests for this?

File: libtenzir/src/tql2/eval_binary.cpp [hunk 2]
```diff
@@ -628,6 +615,53 @@ struct EvalBinOp<Op, string_type, string_type> {
...
+            TENZIR_ASSERT(out->IsValid(0));
+            if (out->Value(0)) {
+              result = true;
+              break;
+            }
+          }
+          check(b.Append(result));
+        }
+      } else {
+        warn("Incompatible types for `in`");
```
Comment (jachris) [inline libtenzir/src/tql2/eval_binary.cpp:656]: As discussed in our call: Diagnostics are always lowercase, and you might want to add the type kinds here.

File: tenzir/integration/data/inputs/json/lists.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{ "x": "", "y": [] }
+{ "x": "", "y": ["Non-Empty", ""] }
+{ "x": "Different", "y": ["1ms", "10d", "111y"] }
+{ "x": 1, "y": [1, 10, 111] }
+{ "x": 1.0, "y": [1, 10, 111] }
+{ "x": 1.0, "y": [1, 10.0, 111] }
```
Comment (jachris) [inline tenzir/integration/data/inputs/json/lists.json:6]: Please add some tests with `null` here.

**Review by jachris — APPROVED** [review]
Thanks. This seems good to go! :rocket:
